controllers/provider-aws/pkg/infrastructure/actuator.go

```diff
@@ -0,0 +1,240 @@
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type actuator struct {
+	config     *rest.Config
+	client     client.Client
+	kubernetes kubernetes.Interface
+	scheme     *runtime.Scheme
+	logger     logr.Logger
+	decoder    runtime.Decoder
+	encoder    runtime.Encoder
+}
+
+// NewActuator creates a new Actuator that updates the status of the handled Infrastructure resources.
+func NewActuator() infrastructure.Actuator {
+	return &actuator{logger: log.Log.WithName("infrastructure-aws-actuator")}
+}
+
+func (c *actuator) InjectScheme(scheme *runtime.Scheme) error {
+	c.scheme = scheme
+	c.decoder = serializer.NewCodecFactory(c.scheme).UniversalDecoder()
+	c.encoder = serializer.NewCodecFactory(c.scheme).EncoderForVersion(c.encoder, extensionsv1alpha1.SchemeGroupVersion)
+	return nil
+}
+
+func (c *actuator) InjectConfig(config *rest.Config) error {
+	c.config = config
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	c.kubernetes = clientset
+	return nil
+}
+
+func (c *actuator) InjectClient(client client.Client) error {
+	c.client = client
+	return nil
+}
+
+func (c *actuator) Exists(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure) (bool, error) {
+	return infrastructure.Status.LastOperation != nil, nil
+}
+
+func (c *actuator) Create(ctx context.Context, config *extensionsv1alpha1.Infrastructure) error {
+	return c.reconcile(ctx, config)
+}
+
+func (c *actuator) Update(ctx context.Context, config *extensionsv1alpha1.Infrastructure) error {
+	return c.reconcile(ctx, config)
+}
+
+func (c *actuator) Delete(ctx context.Context, config *extensionsv1alpha1.Infrastructure) error {
+	return c.delete(ctx, config)
+}
+
+func (c *actuator) reconcile(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure) error {
+	infrastructureConfig := &v1alpha1.InfrastructureConfig{}
+	_, _, err := c.decoder.Decode(infrastructure.Spec.ProviderConfig.Raw, nil, infrastructureConfig)
+	if err != nil {
+		c.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "could not decode providerConfig", 10, err)
+		return err
+	}
+
+	var (
+		namespace = infrastructure.Spec.SecretRef.Namespace
+		name      = infrastructure.Spec.SecretRef.Name
+	)
+
+	providerSecret := &corev1.Secret{}
+	if err := c.client.Get(ctx, kutil.Key(namespace, name), providerSecret); err != nil {
+		return err
+	}
+	var (
+		createVPC         = true
+		vpcID             = "${aws_vpc.vpc.id}"
+		internetGatewayID = "${aws_internet_gateway.igw.id}"
+		dhcpDomainName    = "ec2.internal"
+		vpcCIDR           = infrastructureConfig.Networks.VPC.CIDR
+		region            = infrastructure.Spec.Region
+	)
+
+	awsClient := aws.NewClient(string(providerSecret.Data[awstypes.AccessKeyID]), string(providerSecret.Data[awstypes.SecretAccessKey]), region)
```

Comment:
Instead of using the `aws` client within Gardener, can you please move it to this repo and call it here?